File: zig/armstrong-numbers/armstrong_numbers.zig
const std = @import("std");
const math = @import("std").math;
pub fn isArmstrongNumber(num: u128) bool {
    var buffer: [64]u8 = undefined;
    const str = std.fmt.bufPrint(&buffer, "{d}", .{num}) catch unreachable;
    var sum: u128 = 0;

    for (str) |c| {
        const digit = c - '0';
        sum += math.powi(u128, digit, str.len) catch unreachable;
    }

    return sum == num;
}

File: python/armstrong-numbers/armstrong_numbers.py
def is_armstrong_number(number: int) -> bool:
    num_str = str(number)
    num_len = len(num_str)
    if num_len == 1:
        return True

    return sum(int(char_num) ** num_len for char_num in list(num_str)) == number
